drivers/i2c/i2c_common.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Logging of I2C messages
+ *
+ * Copyright Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#include <drivers/i2c.h>
+
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c);
+
+void i2c_dump_msgs(const char *name, const struct i2c_msg *msgs,
+		   uint8_t num_msgs, uint16_t addr)
+{
+	char data[80], *ptr, *end = data + sizeof(data) - 3;
```

Comment:
How the '80' has been chosen? (is it enough, are there cases where it would require more?)

Reply:
Yes if you have > 26 bytes, but that is pretty rate and in those cases you probably don't want to see everything.

I could adjust the code to support it.

Reply:
Not needed now